### ISSUE TITLE
problem with ssh-tar

### ISSUE THREAD

**Issue by pfuntner** [pfuntner/toys#101]
I created `ssh-tar` recently and it works for simple trees

```
$ ssh-tar -q ubuntu /dev/null | wc
      0       3   10240
$ ssh-tar -q ubuntu /dev/null | tar -tvf -
crw-rw-rw- root/root       1,3 2022-12-13 06:47 dev/null
$
```

but not complex trees:
```
$ ssh-tar -q ubuntu sto/repos/cloud9-docs/site | wc
  97491  439034 12065339
$ ssh-tar -q ubuntu sto/repos/cloud9-docs/site | tar -tf -
sto/repos/cloud9-docs/site/
sto/repos/cloud9-docs/site/assets/
sto/repos/cloud9-docs/site/assets/images/
sto/repos/cloud9-docs/site/assets/images/favicon.png
tar: Skipping to next header
tar: Exiting with failure status due to previous errors
$
```

I'm not sure what's wrong but I would like to get to the bottom of it.

I think it would be a good exercise to create the tarball via `ssh-tar` and `tar` and compare the two.

**Comment by pfuntner** [comment]
This might be an example of binary data and the need for base64-encoding.  If I use `tar` directly with `ssh` and use `base64` to encode the output, the command works fine:

```
$ ssh vm1 'tar -c /usr/bin | base64' | base64 -d | tar -t | headtail
tar: Removing leading `/' from member names
       1 usr/bin/
       2 usr/bin/on_ac_power
       3 usr/bin/hp-makeuri
       4 usr/bin/canberra-gtk-play
       5 usr/bin/pinentry
tar: Removing leading `/' from hard link targets
         .
         .
         .
    1331 usr/bin/xlogo
    1332 usr/bin/bluemoon
    1333 usr/bin/setsid
    1334 usr/bin/seq
    1335 usr/bin/gdb
$
```

There are two ways to go:
- Get rid of `ssh-tar` completely - everything can be done in existing commands, although it's a little inconvenient
- Beef up `ssh-tar` with base64-encoding:
    - Before the output leaves the target, it will be encoded
    - When the output arrives at the source, it will be decoded

**Comment by pfuntner** [comment]
After looking into a bit, I see the tool is already using base64-encoding along with compression.  I'm sure there's a problem at all if you handle the output properly:

```
$ time ssh vm1 'tar -zc /usr/bin | base64' | wcsum
Started 2024-03-09 08:15:08.938334
tar: Removing leading `/' from member names
tar: Removing leading `/' from hard link targets
Stopped 2024-03-09 08:20:01.313480, rc=0
real	0:04:52.375146
user	0:00:00.350000
sys	0:00:00.220000
1094205  1094205 84253741 9cdd91e5b50817e64bf9aec4625b6a04 -
$ time ssh-tar vm1 /usr/bin | wcsum
Started 2024-03-09 08:20:36.454070
tar: Removing leading `/' from member names
tar: Removing leading `/' from hard link targets
Stopped 2024-03-09 08:25:31.702186, rc=0
real	0:04:55.248116
user	0:00:00.350000
sys	0:00:00.320000
1094205  1094205 84253741 9cdd91e5b50817e64bf9aec4625b6a04 -
$ 
```

**Comment by pfuntner** [comment]
From the original symptoms:

```
$ ssh-tar -q ubuntu sto/repos/cloud9-docs/site | tar -tf -
```

This looks like incorrect usage and I'm not sure how it even worked as well as it did... perhaps I changed the tool and improved it.  Not only is the tar ball not uncompressed but the output is not base64-decoded.

**Comment by pfuntner** [comment]
It looks like I fixed the problem in c36c614 in January 2023 which was a couple of months after the initial observation.
